src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ProcedureRequestMapper.java

```diff
@@ -1,12 +1,67 @@
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.FreeTextSegment;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.ReportStatusCode;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.PatientClinicalInfo;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
 public class ProcedureRequestMapper {
-    public ProcedureRequest mapToProcedureRequest(final Message message) {
-        return new ProcedureRequest();
+    private final UUIDGenerator uuidGenerator;
+    private final Map<ReportStatusCode, ProcedureRequest.ProcedureRequestStatus> statusCodeMapping = Map.of(
+        ReportStatusCode.UNSPECIFIED, ProcedureRequest.ProcedureRequestStatus.UNKNOWN
+    );
+
+    public Optional<ProcedureRequest> mapToProcedureRequest(final Message message) {
+        return message.getServiceReportDetails().getSubject().getClinicalInfo()
+            .map(this::mapPatientClinicalInfo);
+    }
+
+    private ProcedureRequest mapPatientClinicalInfo(final PatientClinicalInfo patientClinicalInfo) {
+        final ProcedureRequest procedureRequest = new ProcedureRequest();
+
+        mapFreeText(patientClinicalInfo, procedureRequest);
+        mapStatus(patientClinicalInfo, procedureRequest);
+        procedureRequest.setIntent(ProcedureRequest.ProcedureRequestIntent.NULL);
+        procedureRequest.setCode(new CodeableConcept().setText("unknown"));
+        procedureRequest.setId(uuidGenerator.generateUUID());
+
+        return procedureRequest;
+    }
+
+    private void mapFreeText(final PatientClinicalInfo patientClinicalInfo, final ProcedureRequest procedureRequest) {
+        final List<Annotation> annotations = patientClinicalInfo.getFreeTexts().stream()
+            .map(FreeTextSegment::getTexts)
+            .map(texts -> {
+                final var text = MappingUtils.unescape(String.join(":", texts));
+                return new Annotation().setText(text);
+            })
+            .collect(Collectors.toList());
+
+        if (annotations.isEmpty()) {
+            throw new FhirValidationException("Unable to map message. "
+                + "The FreeText segment is mandatory in Clinical Information");
+        }
+
+        procedureRequest.setNote(annotations);
+    }
+
+    private void mapStatus(final PatientClinicalInfo patientClinicalInfo, final ProcedureRequest procedureRequest) {
+        Optional.ofNullable(patientClinicalInfo)
+            .ifPresent(n -> procedureRequest.setStatus(statusCodeMapping.get(
+                ReportStatusCode.fromCode(patientClinicalInfo.getCode().getCode()))));
```

Comment:
In this case, the `ReportStatusCode` class will throw an exception saying that: `No Report Status Code for 'wrongCode'` because at least for now the ReportStatusCode can also only handle one "UN" code. Do you think that we should implement some additional functionality for this `mapStatus` method?

Reply:
Either way we'll need a test for it! It's a `ClinicalInformationCode` from what I can see, and it doesn't restrict itself to `+UN`

Reply:
Will this be ok? 1f9cf5cfeaa2de95a8c797ed0223be051f48d51b
Thanks!

Reply:
Looks good